src/functions.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace PhpFp;
+
+use PhpFp\Combinators\Curry;
+
+/**
+ * compose :: (b -> c), (a -> b) -> a -> c
+ */
+function compose(callable $f, callable $g): callable
+{
+    return static function ($x) use ($f, $g) {
+        return $f($g($x));
+    };
+}
+define('compose', '\PhpFp\compose');
```

Comment:
@Tohmua probably not a bad idea. I split them to make it clear what the steps involved were.